### PR TITLE
[Added] [BEL-16490] Add Tax Declarations support

### PR DESCRIPTION
**Pull request by blv-albertmaristany**
:question: What
---
- Add Tax Declarations support

:speech_balloon: Why
---
- We are adding the new tax declarations product
- Some testing/linting packages updated

### CODE REVIEW

**Review by blv-dominik — APPROVED** [review]
OK, this was a longer one - but looks good!

File: belvo/resources/tax_declarations.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class TaxDeclarations(Resource):
+    endpoint = "/api/tax-declarations/"
+
+    def create(
+        self,
+        link: str,
+        year_from: str,
+        year_to: str,
+        *,
```
Comment (blv-geoffreyvandercruyssen) [inline belvo/resources/tax_declarations.py:14]: What does this mean?

File: tests/unit/test_tax_declarations.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        "/api/tax-declarations/",
+        data={
+            "link": "fake-link-uuid",
+            "year_from": "2019",
+            "year_to": "2019",
+            "attach_pdf": True,
+            "save_data": True,
+        },
+        raise_exception=False,
+    )
```
Comment (blv-geoffreyvandercruyssen) [inline tests/unit/test_tax_declarations.py:21]: Are we good with only one test or should we try and do the same as in the JS sdk for example?

Comment (blv-albertmaristany) [reply tests/unit/test_tax_declarations.py:21]: If you check the implementation, we are only overwriting the `create` method, while we keep the inherited ones untouched, which are already tested. And, to be honest, just followed the other endpoints doing :slightly_smiling_face:

File: belvo/resources/tax_declarations.py (hunk 1 above)
Comment (blv-albertmaristany) [reply belvo/resources/tax_declarations.py:14]: It separates mandatory arguments from optional ones. It allows you to provide more args in a transparent way.

**Pull request merged**